ZAPD/ZCutscene.cpp

```diff
@@ -994,23 +990,21 @@ size_t CutsceneCommandTextbox::GetCommandSize()
 
 ActorAction::ActorAction(const std::vector<uint8_t>& rawData, uint32_t rawDataIndex)
 {
-	const uint8_t* data = rawData.data();
-
-	action = (uint16_t)BitConverter::ToInt16BE(data, rawDataIndex + 0);
-	startFrame = (uint16_t)BitConverter::ToInt16BE(data, rawDataIndex + 2);
-	endFrame = (uint16_t)BitConverter::ToInt16BE(data, rawDataIndex + 4);
-	rotX = (uint16_t)BitConverter::ToInt16BE(data, rawDataIndex + 6);
-	rotY = (uint16_t)BitConverter::ToInt16BE(data, rawDataIndex + 8);
-	rotZ = (uint16_t)BitConverter::ToInt16BE(data, rawDataIndex + 10);
-	startPosX = BitConverter::ToInt32BE(data, rawDataIndex + 12);
-	startPosY = BitConverter::ToInt32BE(data, rawDataIndex + 16);
-	startPosZ = BitConverter::ToInt32BE(data, rawDataIndex + 20);
-	endPosX = BitConverter::ToInt32BE(data, rawDataIndex + 24);
-	endPosY = BitConverter::ToInt32BE(data, rawDataIndex + 28);
-	endPosZ = BitConverter::ToInt32BE(data, rawDataIndex + 32);
-	normalX = BitConverter::ToFloatBE(data, rawDataIndex + 36);
-	normalY = BitConverter::ToFloatBE(data, rawDataIndex + 40);
-	normalZ = BitConverter::ToFloatBE(data, rawDataIndex + 44);
+	action = (uint16_t)BitConverter::ToInt16BE(rawData, rawDataIndex + 0);
+	startFrame = (uint16_t)BitConverter::ToInt16BE(rawData, rawDataIndex + 2);
+	endFrame = (uint16_t)BitConverter::ToInt16BE(rawData, rawDataIndex + 4);
+	rotX = (uint16_t)BitConverter::ToInt16BE(rawData, rawDataIndex + 6);
+	rotY = (uint16_t)BitConverter::ToInt16BE(rawData, rawDataIndex + 8);
```

Comment:
This doesn't look directly done by you but if you could just call the IntU function that would be better I think.